prefect_azure/blob_storage.py

```diff
@@ -177,3 +191,509 @@ def example_blob_storage_list_flow():
+                            raise RuntimeError(
+                                f"An error occurred when atempting to download from container {self.container_name}: {exc.reason}"
+                            ) from exc
+        return full_container_path
+
+    @sync_compatible
+    async def get_directory(
+        self, from_path: str = None, local_path: str = None
+    ) -> None:
+        """
+        Downloads the contents of a direry from the blob storage to a local path.
+
+        Used to enable flow code storage for deployments.
+
+        Args:
+            from_path: The path of the directory in the blob storage.
+            local_path: The local path where the directory will be downloaded.
+        """
+        await self.download_folder_to_path(from_path, local_path)
+
+    @sync_compatible
+    async def put_directory(
+        self, local_path: str = None, to_path: str = None, ignore_file: str = None
+    ) -> None:
+        """
+        Uploads a directory to the blob storage.
+
+        Used to enable flow code storage for deployments.
+
+        Args:
+            local_path: The local path of the directory to upload. Defaults to
+                current directory.
+            to_path: The destination path in the blob storage. Defaults to
+                root directory.
+            ignore_file: The path to a file containing patterns to ignore
+                during upload.
+        """
+        to_path = "" if to_path is None else to_path
+
+        if local_path is None:
+            local_path = "."
+
+        included_files = None
+        if ignore_file:
+            with open(ignore_file, "r") as f:
+                ignore_patterns = f.readlines()
+
+            included_files = filter_files(local_path, ignore_patterns)
+
+        for local_file_path in Path(local_path).expanduser().rglob("*"):
+            if (
+                included_files is not None
+                and str(local_file_path.relative_to(local_path)) not in included_files
+            ):
+                continue
+            elif not local_file_path.is_dir():
+                remote_file_path = Path(to_path) / local_file_path.relative_to(
+                    local_path
+                )
+                with open(local_file_path, "rb") as local_file:
+                    local_file_content = local_file.read()
+
+                await self.write_path(
+                    remote_file_path.as_posix(), content=local_file_content
+                )
+
+    @sync_compatible
+    async def read_path(self, path: str) -> bytes:
+        """
+        Reads the contents of a file at the specified path and returns it as bytes.
+
+        Used to enable results storage.
+
+        Args:
+            path: The path of the file to read.
+
+        Returns:
+            The contents of the file as bytes.
+        """
+        file_obj = BytesIO()
+        await self.download_object_to_file_object(path, file_obj)
+        return file_obj.getvalue()
```

Comment:
this worked well too. just adding all the examples I use in case that's helpful.
```python
credentials = AzureBlobStorageCredentials(
    connection_string=my_connection_string,
)
container = AzureBlobStorageContainer(
    container_name=existing_container,
    credentials=credentials,
)
my_bytes = container.read_path(path="new_folder")
```